CMake/CreateArmaConfigInfo.cmake

```diff
@@ -10,35 +10,12 @@ else()
   set(OLD_FILE_CONTENTS "")
 endif()
 
-# If we are using Armadillo 5+, ARMA_64BIT_WORD is implicitly enabled.
-set(ARMA_HAS_64BIT_WORD 0) # This may be unnecessary.
-if(NOT (${ARMADILLO_VERSION_MAJOR} LESS 5))
-  # ARMA_64BIT_WORD is only set if we are on a 64-bit system.
-  if (CMAKE_SIZEOF_VOID_P EQUAL 8)
-    set(ARMA_HAS_64BIT_WORD 1)
-  else ()
-    set(ARMA_HAS_64BIT_WORD 0)
-  endif ()
-else()
-  # Otherwise, we'll need to open the config.hpp we are using and inspect the
-  # setting of ARMA_64BIT_WORD.
-  if(EXISTS "${ARMADILLO_INCLUDE_DIR}/armadillo_bits/config.hpp")
-    file(READ "${ARMADILLO_INCLUDE_DIR}/armadillo_bits/config.hpp"
-        ARMA_CONFIG)
 
-    # Extract ARMA_64BIT_WORD.
-    string(REGEX MATCH
-        "[\r\n][ ]*#define ARMA_64BIT_WORD"
-        ARMA_HAS_64BIT_WORD_PRE
-        "${ARMA_CONFIG}")
+# I will set this variable to 0 when cross compile for 386
+# or an embedded system with 32bit, otherwise everything is 64
+# set(ARMA_HAS_64BIT_WORD 0) 
 
-    string(LENGTH "${ARMA_HAS_64BIT_WORD_PRE}" ARMA_HAS_64BIT_WORD)
-  else()
-    # Assumes ARMA_64BIT_WORD is not set.
-    message(WARNING "Armadillo configuration file
-        (${ARMADILLO_INCLUDE_DIR}/armadillo_bits/config.hpp) does not exist!")
-  endif()
-endif()
+set(ARMA_HAS_64BIT_WORD 1)
```

Comment:
I'm a _little_ nervous about this. If not read from the config.hpp file, then maybe an optional argument to set to 32bit?

Reply:
I agree, maybe I can keep the check for the system if it is 64 or 32 bit, even I believe that there are no longer 32-bit machine that can compile mlpack

Reply:
Agreed, this is probably not good to remove.  Probably keeping the check `if (CMAKE_SIZEOF_VOID_P EQUAL 8)` should be sufficient here.  If you want to see what this is used for, take a look at `src/mlpack/core/util/arma_config_check.hpp`. :+1: